javascripts/services/ContactService.js

```diff
@@ -0,0 +1,32 @@
+"use strict"; 
+
+app.service("ContactService", function($http, $q, FIREBASE_CONFIG){
+
+    const addNewContact = (contact) => {
+        return $http.post(`${FIREBASE_CONFIG.databaseURL}/contacts.json`, JSON.stringify(contact));
+    };
+
+    const getFbContacts = (userUid) => {
+        let contacts = [];
+        return $q((resolve, reject) => {
+            $http.get(`${FIREBASE_CONFIG.databaseURL}/contacts.json?orderBy="uid"&equalTo="${userUid}"`).then((results) => {
+                let fbContacts = results.data;
+                Object.keys(fbContacts).forEach((key) => {
+                    fbContacts[key].id = key;
+                    contacts.push(fbContacts[key]);
+                    resolve(contacts);                    
```

Comment:
this resolution for your contact should be outside of your looping over of your Object.keys